core/kogitoservice/deployer_resources.go

```diff
@@ -44,6 +45,28 @@ func (s *serviceDeployer) createRequiredResources() (resources map[reflect.Type]
 		resources[reflect.TypeOf(imgv1.ImageStream{})] = []resource.KubernetesResource{imageStream}
 	}
 
+	appProps := map[string]string{}
+	appPropsConfigMapHandler := NewAppPropsConfigMapHandler(s.Context)
+	if len(s.instance.GetSpec().GetPropertiesConfigMap()) > 0 {
+		s.Log.Debug("custom app properties are provided in custom properties ConfigMap", "PropertiesConfigMap", s.instance.GetSpec().GetPropertiesConfigMap())
+		propertiesConfigMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: s.instance.GetNamespace(),
+				Name:      s.instance.GetSpec().GetPropertiesConfigMap(),
+			},
+		}
+		if exists, err := kubernetes.ResourceC(s.Client).Fetch(propertiesConfigMap); err != nil {
+			return resources, err
+		} else if exists {
```

Comment:
What should we do in case user actually defines config map but it doesn't exist?

I am thinking whether we should rather consider this as an error. Such issue could be caused for example by having a typo in the config map name. 
In case operator act like there is an empty config map the user can be mislead in thinking that everything is fine.

Reply:
Good question. I think we should fail, hence updating the statusCondition with a meaningful message so users can easily fix it.

Reply:
I changed it to throw an error when it can't find the propertiesConfigMap now and added a unit test for it. It'll continually throw the error in the operator log and won't deploy the pod until its fixed, so it's an obvious error for the user now.